mindmeld/components/question_answerer.py

```diff
@@ -797,13 +826,22 @@ def build_query(self):
                         }
                     }
                 else:
-                    clause = {
-                        "match": {
-                            self.field + ".normalized_keyword": {
-                                "query": self.value
+                    if self.query_type == 'text':
+                        clause = {
+                            "match": {
+                                self.field + ".char_ngram": {
+                                    "query": self.value
+                                }
+                            }
+                        }
+                    else:
```

Comment:
check whether the query_type is keyword, otherwise throw UnsupportedQueryTypeException